app/src/main/java/com/stevesoltys/seedvault/restore/install/ApkRestore.kt

```diff
@@ -37,15 +38,22 @@ internal class ApkRestore(
 ) {
 
     private val pm = context.packageManager
+    private val restrictions = context.getSystemService(UserManager::class.java).userRestrictions
 
     @Suppress("BlockingMethodInNonBlockingContext")
     fun restore(backup: RestorableBackup) = flow {
+        val cantInstallApks = restrictions.getBoolean(UserManager.DISALLOW_INSTALL_APPS,
+            false) ||
+            restrictions.getBoolean(UserManager.DISALLOW_INSTALL_UNKNOWN_SOURCES,
+            false) ||
+            restrictions.getBoolean(UserManager.DISALLOW_INSTALL_UNKNOWN_SOURCES_GLOBALLY,
+            false)
```

Comment:
Need to double check if this is how we want to check for it - (false is defaultValue)

Reply:
That is a bit hard to read. Maybe define it as an extension function on `userRestrictions`, so you can mock its return value for testing?

Reply:
Could also statically import the `UserManager.*` constants.